resolver-dns/src/main/java/io/netty/resolver/dns/DnsNameResolver.java

```diff
@@ -160,7 +160,9 @@ private static boolean anyInterfaceSupportsIpV6() {
                 NetworkInterface iface = interfaces.nextElement();
                 Enumeration<InetAddress> addresses = iface.getInetAddresses();
                 while (addresses.hasMoreElements()) {
-                    if (addresses.nextElement() instanceof Inet6Address) {
+                    InetAddress inetAddress = addresses.nextElement();
+                    if (inetAddress instanceof Inet6Address && !inetAddress.isAnyLocalAddress() &&
+                        !inetAddress.isLoopbackAddress() && !inetAddress.isLinkLocalAddress()) {
```

Comment:
@minwoox hmm... shouldn't we still support AAAA if its a loopback address ? I mean the user may run the DNS server on the same machine.

Reply:
So you mean the client and the server in the same machine?
Yes, that could be possible, but I think, in that case, there should be another valid address that the client can connect to the server so that the client can connect to other endpoints using that address.

Reply:
yes ... I think it is valid to only have `::1` configured for this case.

Reply:
If the machine does not have any other addresses than `::1`, shouldn't the client send any `AAAA` query? Because the client cannot establish a connection using the resolved IPv6 address to the endpoint? 🤔

Reply:
@minwoox no... the resolver is used during bootstrap but may also used separately (for only resolving addresses). It doesn't matter if we can connect to the address or not, thats no the job of the resolver to check.

Reply:
Ah, now I got it what you meant. 😄 
But I still think that setting `DEFAULT_RESOLVE_ADDRESS_TYPES` to `ResolvedAddressTypes.IPV4_ONLY` is safe when there are only local addresses.
Because there's a chance that the `DnsNameResolver` gets the `AAAA` response ahead of `A` response and if the resolver uses it, the client would not make a connection.

If a user wants to do send an IPv6 query like you gave an example, he or she can just set `IPV4_PREFERRED` or `IPV6_PREFERRED` when building `DnsNameResovler`.
What do you think?

Reply:
True...  I think the title of the PR did confuse me as I did not realise this is just about default behaviour

Reply:
Ah sorry about it. 😅 Just use the title from the original PR without much thinking. 
https://github.com/line/armeria/pull/2464 Let me change it. Thank you!